File: usecases/src/test/java/com/zeyad/usecases/mapper/DAOMapperTest.java
package com.zeyad.usecases.mapper;

import com.google.gson.internal.LinkedTreeMap;
import com.zeyad.usecases.TestRealmModel;

import junit.framework.Assert;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.ArrayList;
import java.util.Collections;

/**
 * @author by ZIaDo on 5/14/17.
 */
public class DAOMapperTest {
    private DAOMapper daoMapper;

    @Before
    public void setUp() {
        daoMapper = new DAOMapper();
    }

    @After
    public void tearDown() {
        daoMapper = null;
    }

    @Test
    public void mapTo() {
        Assert.assertEquals(
                daoMapper.mapTo(new Object(), Object.class).getClass(), LinkedTreeMap.class);
        Assert.assertEquals(
                daoMapper.mapTo(new Object(), TestRealmModel.class).getClass(),
                TestRealmModel.class);
    }

    @Test
    public void mapAllTo() {
        Assert.assertEquals(
                daoMapper.mapAllTo(Collections.EMPTY_LIST, Object.class).getClass(),
                ArrayList.class);
        Assert.assertEquals(
                daoMapper
                        .mapAllTo(Collections.singletonList(new Object()), Object.class)
                        .getClass(),
                ArrayList.class);
        Assert.assertEquals(
                daoMapper
                        .mapAllTo(
                                Collections.singletonList(new TestRealmModel()),
                                TestRealmModel.class)
                        .getClass(),
                ArrayList.class);
    }
}

File: usecases/src/test/java/com/zeyad/usecases/mapper/DAOMapperTest.kt
package com.zeyad.usecases.mapper

import com.google.gson.internal.LinkedTreeMap
import com.zeyad.usecases.TestModel
import junit.framework.Assert
import org.junit.After
import org.junit.Before
import org.junit.Test
import java.util.*

/**
 * @author by ZIaDo on 5/14/17.
 */
class DAOMapperTest {
    private var daoMapper: DAOMapper? = null

    @Before
    fun setUp() {
        daoMapper = DAOMapper()
    }

    @After
    fun tearDown() {
        daoMapper = null
    }

    @Test
    fun mapTo() {
        Assert.assertEquals(
                daoMapper!!.mapTo<Any>(Any(), Any::class.java).javaClass, LinkedTreeMap::class.java)
        Assert.assertEquals(
                daoMapper!!.mapTo<Any>(Any(), TestModel::class.java).javaClass,
                TestModel::class.java)
    }

    @Test
    fun mapAllTo() {
        Assert.assertEquals(
                daoMapper!!.mapAllTo<Any>(Collections.EMPTY_LIST, Any::class.java).javaClass,
                ArrayList::class.java)
        Assert.assertEquals(
                daoMapper!!
                        .mapAllTo<Any>(listOf(Any()), Any::class.java).javaClass,
                ArrayList::class.java)
        Assert.assertEquals(
                daoMapper!!
                        .mapAllTo<Any>(
                                listOf(TestModel()),
                                TestModel::class.java).javaClass,
                ArrayList::class.java)
    }
}
